src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -136199,5 +136199,74 @@ void M()
                 //             _ = null,
                 Diagnostic(ErrorCode.ERR_DiscardTypeInferenceFailed, "_").WithLocation(10, 13));
         }
+
+        [Fact]
+        [WorkItem(47221, "https://github.com/dotnet/roslyn/issues/47221")]
+        public void PropertyAccessorWithNullableContextAttribute_01()
+        {
+            var source0 =
+@"#nullable enable
+public class A
+{
+    public object? this[object x, object? y] => null;
+    public static A F(object x) => new A();
+    public static A F(object x, object? y) => new A();
+}";
+            var comp = CreateCompilation(source0);
+            var ref0 = comp.EmitToImageReference();
+
+            var source1 =
+@"#nullable enable
+class B
+{
+    static void F(object x, object? y)
+    {
+        var a = A.F(x, y);
+        var b = a[x, y];
+        b.ToString(); // 1
+    }
+}";
+            comp = CreateCompilation(source1, references: new[] { ref0 });
+            comp.VerifyEmitDiagnostics(
+                // (8,9): warning CS8602: Dereference of a possibly null reference.
+                //         b.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "b").WithLocation(8, 9));
+        }
+
+        [Fact]
+        [WorkItem(47221, "https://github.com/dotnet/roslyn/issues/47221")]
+        public void PropertyAccessorWithNullableContextAttribute_02()
+        {
+            var source0 =
+@"#nullable enable
+public class A
+{
+    public object this[object? x, object y] => new object();
+    public static A? F0;
```

Comment:
Are F0-4 necessary for this test?

Reply:
> Are F0-4 necessary for this test?

Yes, we need more annotated reference types than unannotated reference types in the type definition to generate a `[NullableContext(2)]` on type, and that attribute is needed to force a `[NullableContext(1)]` on the property accessor.

Reply:
> Yes, we need more annotated reference types than unannotated reference types in the type definition

that's probably why I couldn't repro this standalone myself....